### PR TITLE
[WIP] OS Notification on update downloaded and minor refactor

### PR DESCRIPTION
**Pull request by AlphaStyle**
fix [165](https://github.com/hql287/Manta/issues/165)

*Enabling* and *disabling* OS Notification requires restart.
Moved `center-on-primary-display.js` and `notify.js` to a shared `helpers` folder in the *root directory*.

@hql287 Could you test if this actually works? Anything you would like to improve?

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -185,6 +185,7 @@ function setInitialValues() {
       language: 'en',
       sound: 'default',
       muted: false,
+      notification: true,
```
Comment (hql287) [inline app.js]: I think this is not necessary. The notification should always be on, don't you think?

File: helpers/notify.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const appConfig = require('electron-settings');
+
+const Notify = options => {
+  const permission = appConfig.get('general.notification');
```
Comment (hql287) [inline helpers/notify.js]: We also should remove the conditional check here

File: main/updater.js [hunk 3]
```diff
@@ -69,6 +71,7 @@ autoUpdater.on('download-progress', progressObj => {
 
 // Update Downloaded
 autoUpdater.on('update-downloaded', info => {
+  Notify({ title: 'Manta Update', body: 'Updates has been downloaded' });
```
Comment (hql287) [inline main/updater.js]: This needs translation. Please refers to PR #187 for implementation detail of i18next

File: app.js [hunk 4]
```diff
@@ -239,6 +240,7 @@ function migrateData() {
           language: appSettings.language,
           sound: appSettings.sound,
           muted: appSettings.muted,
+          notification: appSettings.notification,
```
Comment (hql287) [inline app.js]: Also, this is actually not needed as this is a new value.

**Comment by hql287** [comment]
This should be an easy fix and I figured it would be easier to make a new PR, plus I haven't heard back from you so I'm closing this in favour of #201.

**Comment by AlphaStyle** [comment]
I am sorry for not responding! I know it is somewhat rude to start a PR and then dont follow up, and I appolegies for this. Most of the time I can only code during the weekends.

**Pull request closed without merging**